[Justin Hoe] iP

Justinhoejj opened the pull request:
### Duke Personal Assistant

> OK -Jerry

- [Here](https://github.com/Justinhoejj/ip)
- Is a
- Bullet List 
_yeppers_

1. Here's
2. Some
3. Numbered List

this is **verry bold** !!

```java
public class Main {
    public static void main() {
        // dont do nothing
    }
}
```

- [ ] 🌧 
- [ ] 🌞 

~~nice~~

Commit:
Add Gradle support

Justinhoejj committed:
Implement Greet, Echo, Exit

Justinhoejj committed:
Implement Add, List

Justinhoejj committed:
Mark as Done

Justinhoejj committed:
Todos, Events, Deadlines, PrintResponse

Justinhoejj committed:
I/O testing

Justinhoejj committed:
Handle errors

Justinhoejj committed:
Delete

Justinhoejj committed:
Implement save

Justinhoejj committed:
Implement deadline with LocalDateTime

Justinhoejj committed:
Merge branch-Level-8 to master

Justinhoejj committed:
Implement abstract Command class

Justinhoejj committed:
Refactor related code as classes

Justinhoejj committed:
Integrate existing code with new classes

Justinhoejj committed:
Organise into packages

Justinhoejj committed:
Implement junit test

Justinhoejj committed:
Turn task to abstract class

Justinhoejj committed:
Include JavaDocs

Justinhoejj committed:
Enforce coding standard

Justinhoejj committed:
Implement find by task name

Justinhoejj committed:
Merge branch 'branch-Level-9' into master

Justinhoejj committed:
merge branch 'branch-A-CodingStandard' into master

Justinhoejj committed:
merge branch 'branch-A-JavaDoc' into master

Justinhoejj committed:
Refactor AddCommand and Parser

Justinhoejj commented:
### Duke Personal Assistant

> OK -Jerry

- [Here](https://github.com/Justinhoejj/ip)
- Is a
- Bullet List 
_yeppers_

1. Here's
2. Some
3. Numbered List

this is **verry bold** !!

```java
public class Main {
    public static void main() {
        // dont do nothing
    }
}
```

- [ ] 🌧 
- [ ] 🌞 

~~nice~~

Justinhoejj committed:
Add gradle files into project

Justinhoejj committed:
Enforce coding standard

Justinhoejj committed:
Implement JavaFX GUI

Justinhoejj committed:
Set duke & user profile picture

Justinhoejj committed:
Refactor UI

lingshanng commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+        switch (taskType) {
+        case "T":
+            return new Todo(done, taskDetails[2]);
+        case "E": {
+            String info = taskDetails[3];
+            return new Event(done, taskDetails[2], LocalDateTime.parse(info));
+        }
+        case "D": {
+            String info = taskDetails[3];
+            return new Deadline(done, taskDetails[2], LocalDateTime.parse(info));
+        }
```
I think the braces around the case can be omitted 
```suggestion
        case "E": 
            String info = taskDetails[3];
            return new Event(done, taskDetails[2], LocalDateTime.parse(info));
        case "D": 
            String info = taskDetails[3];
            return new Deadline(done, taskDetails[2], LocalDateTime.parse(info));
```

Justinhoejj replied:
Thats a good catch I did not notice the code block thanks !

lingshanng commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+     * @param taskNumber first task has task number of 1.
+     * @return task that was marked as done.
+     */
+    public Task markAsDone(int taskNumber) {
```
maybe can give this method a different name eg `markTaskAsDone` to differentiate it from `task.markAsDone()`

lingshanng reviewed (commented):
Looks good overall, I just have a few suggestions.

Jiwei99 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    /**
+     * Gets a duke command based on user input message.
+     * @param input user instructions for duke in form of string.
```
Should leave a line between method description and parameter section as per Coding Standard

Jiwei99 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,57 @@
+        taskList.addTask(task);
+        storage.writeToDisk(taskList.compileTasks());
+        return String.format("Caan Do!\n  added: %s\nLook at me! %d tasks in the list now!",
+                this.task,
+                taskList.getSize());
```
Can combine these 2 lines into 1 line

Jiwei99 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+
+    /**
+     * Constructs task with a deadline.
+     * @param done task is completed.
```
Same as above, can leave a line between method description and parameter section.

Jiwei99 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+     * Instantiates a task list to store tasks.
+     */
+    public TaskList() {
+
```
Can remove the extra space here

Jiwei99 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+ */
+public abstract class Command {
+    /**
+     * Abstract method that performs a series of action to change the state of Duke
+     * and returns a message depending on the command type.
```
Perhaps could shorten the method description, since it is supposed to be a short summary.

Jiwei99 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+
+    /**
+     *
+     * Turn Deadline into a string for saving to memory.
```
Should use `Turns` instead

Jiwei99 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,44 @@
+    private final LocalDateTime dateTime;
+
+    /**
+     * Construct event with date time.
```
Should use `Constructs` instead

Jiwei99 commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,31 @@
+
+    /**
+     *
+     * Turn todo into a string for saving to memory.
```
As mentioned above

Jiwei99 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    /**
+     * Filter tasks in the task list with name that includes the search term.
```
Should use `Filters` instead

Jiwei99 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    /**
+     * Turn Event into a string for saving to memory.
```
Should use `Turns` here instead

Jiwei99 reviewed (commented):
Generally LGTM, the coding standard is well followed. Just some minor JavaDocs changes required, such as using `Turns` instead of `Turn`, and perhaps can further shorten some method descriptions :)

zzybluebell commented on src/main/java/duke/util/Ui.java:
```diff
@@ -15,6 +17,15 @@ public void respond(String message) {
+    }
+
     public String nextCommand() {
         return scanner.nextLine();
```
Overall the coding standard is well followed and good looking

zzybluebell commented on src/test/java/duke/command/DoneCommandTest.java:
```diff
@@ -19,8 +18,7 @@ public void executeDoneCommand_invalidTaskNumber_throwDukeException() {
-            new DoneCommand(taskNumber).execute(storage, tl, ui);
+            new DoneCommand(taskNumber).execute(storage, tl);
         } catch (DukeException e) {
             assertEquals(String.format("Task number %d invalid.", taskNumber), e.getMessage());
```
Overall looks good

zzybluebell reviewed (commented):
Good code standard following, just a bit braces {} omitted in switch case part :)

Justinhoejj committed:
Implement assert

Implementing assert will allow for defining assumptions about the state of a program so that the runtime can verify.
Assertions also point out errors in the code that the programmer might have overlooked.

Justinhoejj committed:
Improve code quality

Improved code quality ensures that the code is more readable, easier to maintain.

- Add constant index values to avoid "magic numbers" in task list.
- Minor changes to styling of javaDocs.

Justinhoejj committed:
Merge pull request #1 from Justinhoejj/branch-A-CodeQuality

Improve code quality

Justinhoejj committed:
Merge branch 'master' into branch-A-Assertions

Justinhoejj committed:
Merge pull request #2 from Justinhoejj/branch-A-Assertions

Branch a assertions

Justinhoejj committed:
Set up CI

Justinhoejj committed:
Revert "Set up CI"

This reverts commit 325ed8e4

Justinhoejj committed:
Set up CI

Justinhoejj committed:
Merge branch 'master' of https://github.com/Justinhoejj/ip into master

Justinhoejj committed:
Implement edit task

Justinhoejj committed:
Merge pull request #3 from Justinhoejj/branch-A-Assertions

Branch a assertions

Justinhoejj committed:
Add duke memory file

Justinhoejj committed:
Merge branch 'master' of https://github.com/Justinhoejj/ip into master

Justinhoejj committed:
Polish GUI

Justinhoejj committed:
Merge pull request #5 from Justinhoejj/branch-A-BetterGui

Polish GUI

Justinhoejj committed:
Add personality

Justinhoejj committed:
Revise test file output

Justinhoejj committed:
Merge pull request #6 from Justinhoejj/branch-A-BetterGui

Add personality to command responses and modify test file to accept new response

Justinhoejj committed:
Fix response bugs

Justinhoejj committed:
Add a user guide

Justinhoejj committed:
Merge pull request #7 from Justinhoejj/branch-A-UserGuide

Branch a user guide

Justinhoejj committed:
Improve user guide

Justinhoejj committed:
Rename Ui.png

Justinhoejj committed:
Merge pull request #8 from Justinhoejj/branch-A-UserGuide

Improve User Guide

Justinhoejj committed:
Edit javadocs

Justinhoejj committed:
Edit exit response

Justinhoejj committed:
Improve application summary

Justinhoejj committed:
Merge branch 'master' into branch-A-UserGuide

Justinhoejj committed:
Update javadoc summary

Justinhoejj committed:
Merge pull request #9 from Justinhoejj/branch-A-UserGuide

Branch a user guide
